### ISSUE TITLE
Unable to reproduce the AUC of FinalNet

### ISSUE THREAD

**Issue by hellowodex** [reczoo/BARS#30]
Here is the config:

{
    "batch_norm": "True",
    "batch_size": "4096",
    "block1_dropout": "0.1",
    "block1_hidden_activations": "ReLU",
    "block1_hidden_units": "[400, 400]",
    "block2_dropout": "0",
    "block2_hidden_activations": "ReLU",
    "block2_hidden_units": "[400, 400]",
    "block_type": "2B",
    "data_format": "csv",
    "data_root": "/data/workspace/rec/dataset/Criteo_x1/",
    "dataset_id": "criteo_x1_fuxictr2",
    "debug_mode": "False",
    "early_stop_patience": "2",
    "embedding_dim": "10",
    "embedding_regularizer": "1e-05",
    "epochs": "100",
    "eval_interval": "1",
    "eval_steps": "None",
    "feature_cols": "[{'active': True, 'dtype': 'float', 'name': ['I1', 'I2', 'I3', 'I4', 'I5', 'I6', 'I7', 'I8', 'I9', 'I10', 'I11', 'I12', 'I13'], 'type': 'numeric'}, {'active': True, 'dtype': 'float', 'name': ['C1', 'C2', 'C3', 'C4', 'C5', 'C6', 'C7', 'C8', 'C9', 'C10', 'C11', 'C12', 'C13', 'C14', 'C15', 'C16', 'C17', 'C18', 'C19', 'C20', 'C21', 'C22', 'C23', 'C24', 'C25', 'C26'], 'type': 'categorical'}]",
    "feature_specs": "None",
    "gpu": "0",
    "group_id": "None",
    "label_col": "{'dtype': 'float', 'name': 'label'}",
    "learning_rate": "0.001",
    "loss": "binary_crossentropy",
    "metrics": "['AUC', 'logloss']",
    "min_categr_count": "1",
    "model": "FinalNet",
    "model_id": "FINAL_criteo_x1_028_b6c861e4",
    "model_root": "./checkpoints/FINAL_criteo_x1/",
    "monitor": "AUC",
    "monitor_mode": "max",
    "net_regularizer": "0",
    "num_workers": "3",
    "optimizer": "adam",
    "ordered_features": "None",
    "pickle_feature_encoder": "True",
    "save_best_only": "True",
    "seed": "2021",
    "shuffle": "True",
    "task": "binary_classification",
    "test_data": "/data/workspace/rec/dataset/Criteo_x1/test.csv",
    "train_data": "/data/workspace/rec/dataset/Criteo_x1/train.csv",
    "use_field_gate": "True",
    "valid_data": "/data/workspace/rec/dataset/Criteo_x1/valid.csv",
    "verbose": "1"
}

Here is the log:
[FINAL_criteo_x1_028_b6c861e4.log](https://github.com/reczoo/BARS/files/14665726/FINAL_criteo_x1_028_b6c861e4.log)

**Comment by zhujiem** [comment]
What are FuxiCTR version and other package versions?

**Comment by hellowodex** [comment]
The version of FuxiCTR is 2.1.3
<img width="251" alt="image" src="https://github.com/reczoo/BARS/assets/19728187/76a22e7c-c57e-488b-8a39-623a891823c1">

**Comment by hellowodex** [comment]
> What are FuxiCTR version and other package versions?
The version of FuxiCTR is 2.1.3
image

Other package versions:
Python : 3.10.13
Torch : 2.2.1+cu121
Numpy : 1.26.4

**Comment by zhujiem** [comment]
I am not sure whether torch v2+ has any modifications that influence the performance. I tested the models using torch 1.10. You could have a test first.
